### DISCUSSION TITLE
Email login verification not working in the web console

### DISCUSSION THREAD

**Discussion by Jono-Moss** [rustdesk/rustdesk#5242] [Q&A] [answered]
Hi, 
This might be a possible bug, I have enabled email login verification
for my user, When I login I receive the email with the code instantly,
But when I enter the code I get the following message:

"The verification code is incorrect or has expired".

I have checked the servers time / time zone. It is set to my local time zone:
Africa/Johannesburg
The time and date is correct on the server.

If I choose the "Forgot password" option, I get emailed a verification code and that code works, it will log me in.

So it seems to be something wrong with the Login verification function.

**Comment by rustdesk** [comment]
Thanks for your report.

@dinger1986 could you confirm if it works for you. It just tested, it did work for me.

**Reply by Jono-Moss** [reply]
Here is a screen recording of the issue.
Sorry it took a bit of time to blur out sensitive info and
I had to blur out a lot.

https://github.com/rustdesk/rustdesk/assets/118931687/34b94c09-1fbb-4d9f-a7ef-ce18e0f15ee1

**Reply by rustdesk** [reply]
Could you invite me?  then I will try to log in.

Apologies that you spend so much time to make this video.

**Reply by Jono-Moss** [reply]
Hi, thank you. I have invited you.

**Reply by rustdesk** [reply]
Interesting, it works for me.

**Reply by rustdesk** [reply]
Both client, and server timezone are Africa/Johannesburg?

**Reply by Jono-Moss** [reply]
Hi, 
okay I have just figured out what is wrong. It is definitely a bug.
So my username on Rustdesk is: jonathan (with a lowercase j)
When I am signing in, I am using "Jonathan" (With an uppercase J) 

When doing this, it allows me to login, but when it is doing the code verification step.
It sees that there is no user "Jonathan" with a Uppercase J.
So the verification fails.

If I login using "jonathan"  (with a lowercase j)
I can login and then the code is then verified.

So there is a bug in the login process.  It should be case sensitive with regards to the username.
At the moment it accepts the username no matter if it is Upper or Lower Case.

I hope this helps.

**Reply by rustdesk** [reply]
Thanks for you figuring this out. I will reproduce and fix.

**Reply by Jono-Moss** [reply]
Yip tested it again on a different laptop and client. 

It is definitely got to do with the username case sensitivity.
If I login with Uppercase J. The verification code doesn't work.
If I login with lower case j. The code works.
The timezone doesn't appear to be the issue.

Thank you for your help and awesome software.

**Reply by rustdesk** [reply, marked as answer]
We will keep the case insensentive, but fix the verification issue, I have pointed out the problem in the code.
